src/commerce-handler.js

```diff
@@ -9,66 +10,140 @@ CommerceHandler.prototype.logCommerceEvent = function(event) {
-                    delete optimizelyEvent.tags.Revenue;
+                window['optimizely'].push(eventCopy);
+            }
+        }
+
+        if (optimizelyFullStackEvents.events[expandedEvent.EventName]) {
+            var eventKey = event.EventName,
+            userId,
+            userAttributes = {},
+            eventTags = {};
+
+            userAttributes = expandedEvent.EventAttributes || {};
+
+            if (window.mParticle && window.mParticle.Identity) {
+                var userIdentities = window.mParticle.Identity.getCurrentUser().getUserIdentities()['userIdentites'];
+    
+                switch(forwarderSettings.userIdField) {
+                    case 'customerId':
+                        userId = userIdentities["customerId"];
+                        break;
+                    case 'email':
+                        userId = userIdentities["email"];
+                        break;
+                    case 'mpid':
+                        userId = userIdentities["mpid"];
+                        break;
+                    case 'other':
+                        userId = userIdentities["other"];
+                        break;
+                    case 'other2':
+                        userId = userIdentities["other2"];
+                        break;
+                    case 'other3':
+                        userId = userIdentities["other3"];
+                        break;
+                    case 'other4':
+                        userId = userIdentities["other4"];
+                        break;
+                    default:
+                        userId = null;
                 }
             }
-        } else {
-            optimizelyEvent.eventName = expandedEvent.EventName;
+
             if (
-                event.CustomFlags &&
-                event.CustomFlags['Optimizely.EventName']
+                event.EventCategory ===
+                    mParticle.CommerceEventType.ProductPurchase ||
+                event.EventCategory === mParticle.CommerceEventType.ProductRefund
             ) {
-                optimizelyEvent.eventName =
-                    event.CustomFlags['Optimizely.EventName'];
-            }
-        }
+                if (expandedEvent.EventName.indexOf('Total') > -1) {
+                        eventKey = expandedEvent.EventName;
 
-        // Events that are added to the OptimizelyUI will be available on optimizelyEvents.events
-        // Ignore events not included in the Optimizely UI
-        if (optimizelyEvents.events[optimizelyEvent.eventName]) {
-            var eventCopy = {};
-            for (var key in optimizelyEvent) {
-                eventCopy[key] = optimizelyEvent[key];
+                    // Overall purchase event
+                    if (
+                        expandedEvent.EventAttributes &&
+                        expandedEvent.EventAttributes['Total Amount']
+                    ) {
+                        eventTags.revenue =
```

Comment:
are there other examples what `eventTags` might have in optimizely? i wonder if it should be a copy of the EventAttributes, as well the reserved `revenue` key

Reply:
Looking at docs again, and you're right, it should. Will add alongside the reserved revenue key.

Reply:
Looks like I was mapping EventAttributes to userAttributes in both commerce-handler and event-handler. Changing this logic to map to eventTags instead.